### ISSUE TITLE
SamplingProbablityCorrection adds log(p)

### ISSUE THREAD

**Issue by r-mirchandani** [tensorflow/recommenders#278]
I've been doing some digging into using `SamplingProbablityCorrection` for a retrieval task and have noticed something a bit weird. The current implementation of probability correction **adds** the log of the sample probability.
This can be seen here: **https://github.com/tensorflow/recommenders/blob/d18f2db8c7ff8651b552d97eac1c63cc379f9c52/tensorflow_recommenders/layers/loss.py#L157**

However, [in the paper that I assume serves as the foundations of this package](https://storage.googleapis.com/pub-tools-public-publication-data/pdf/6c8a86c981a62b0126a11896b7f6ae0dae4c3566.pdf), logits are corrected with the following equation:
![image](https://user-images.githubusercontent.com/19165069/115487271-b548a680-a29b-11eb-923a-5820be7c0eeb.png)

Am I missing something here, or there an error in the implementation?

**Comment by nialloh23** [comment]
@r-mirchandani This is a known issue that I believe is going to be fixed in an upcoming release. There is temporary solution in this linked issue which involves using the reciprocal of the sampling probability:
https://github.com/tensorflow/recommenders/issues/257

**Comment by r-mirchandani** [comment]
@nialloh23 Ah thanks heaps for that! Good to know I'm not just going crazy from staring at the source code - I'll give the workaround a fix and wait for the next release!

**Comment by maciejkula** [comment]
Yep we'll cut a release soon!
